### ISSUE TITLE
RUSTSEC-2021-0020: Bump Substrate for Updated `hyper`

### ISSUE THREAD

**Issue by HCastano** [paritytech/parity-bridges-common#710]
Cargo deny's been working hard, and found another advisory. This time it's related to
`hyper`. In order for Substrate to fix this it will need to update its `jsonrpc-*`
crates from `v15.1` to at least `v16.0`.

Output from `cargo-deny`:

```
error[A001]: Multiple Transfer-Encoding headers misinterprets request payload
    ┌─ /github/workspace/Cargo.lock:244:1
    │
244 │ hyper 0.12.35 registry+https://github.com/rust-lang/crates.io-index
    │ ------------------------------------------------------------------- security vulnerability detected
    │
    = ID: RUSTSEC-2021-0020
    = Advisory: https://rustsec.org/advisories/RUSTSEC-2021-0020
    = hyper's HTTP server code had a flaw that incorrectly understands some requests
      with multiple transfer-encoding headers to have a chunked payload, when it
      should have been rejected as illegal. This combined with an upstream HTTP proxy
      that understands the request payload boundary differently can result in
      "request smuggling" or "desync attacks".
    = Announcement: https://github.com/hyperium/hyper/security/advisories/GHSA-6hfq-h8hq-87mf
    = Solution: Upgrade to >=0.14.3 OR >=0.13.10, <0.14.0

error[A001]: Multiple Transfer-Encoding headers misinterprets request payload
    ┌─ /github/workspace/Cargo.lock:245:1
    │
245 │ hyper 0.13.9 registry+https://github.com/rust-lang/crates.io-index
    │ ------------------------------------------------------------------ security vulnerability detected
    │
    = ID: RUSTSEC-2021-0020
    = Advisory: https://rustsec.org/advisories/RUSTSEC-2021-0020
    = hyper's HTTP server code had a flaw that incorrectly understands some requests
      with multiple transfer-encoding headers to have a chunked payload, when it
      should have been rejected as illegal. This combined with an upstream HTTP proxy
      that understands the request payload boundary differently can result in
      "request smuggling" or "desync attacks".
    = Announcement: https://github.com/hyperium/hyper/security/advisories/GHSA-6hfq-h8hq-87mf
    = Solution: Upgrade to >=0.14.3 OR >=0.13.10, <0.14.0
```

**Comment by HCastano** [comment]
Related Substrate issue: https://github.com/paritytech/substrate/issues/8074

**Comment by svyatonik** [comment]
closed by #980
